hs_core/serialization.py

```diff
@@ -0,0 +1,842 @@
+            coverage = GenericResourceMeta.ResourceCoveragePeriod(str(coverage_lit))
+            self.coverages.append(coverage)
+
+        print("\t\tCoverages: ")
+        for c in self.coverages:
+            print("\t\t\t{0}".format(str(c)))
+
+        # Get relations
+        for s, p, o in self._rmeta_graph.triples((None, rdflib.namespace.DC.relation, None)):
+            for pred, obj in self._rmeta_graph.predicate_objects(o):
+                relation = GenericResourceMeta.ResourceRelation(obj, pred)
+                self.relations.append(relation)
+
+        print("\t\tRelations: ")
+        for r in self.relations:
+            print("\t\t\t{0}".format(str(r)))
+
+        # Get sources
+        for s, p, o in self._rmeta_graph.triples((None, rdflib.namespace.DC.source, None)):
+            for pred, obj in self._rmeta_graph.predicate_objects(o):
+                source = GenericResourceMeta.ResourceSource(obj, pred)
+                self.sources.append(source)
+
+        print("\t\tSources: ")
+        for r in self.sources:
+            print("\t\t\t{0}".format(str(r)))
+
+    def get_owner_uri(self):
+        """
+        Return the creator with the lowest order.
+
+        :return: String representing the URI of the owner of the resource.
+        """
+        min_order = sys.maxint
+        owner_uri = None
+        for c in self.creators:
+            if c.order < min_order:
+                min_order = c.order
+                owner_uri = c.uri
+                break
+        return owner_uri
+
+    def write_metadata_to_resource(self, resource):
+        """
+        Write metadata to resource
+
+        :param resource: HydroShare resource instance
+        """
+        if self.abstract:
+            resource.metadata.create_element('description', abstract=self.abstract)
+        if self.rights:
+            try:
+                resource.metadata.create_element('rights', statement=self.rights.statement)
+            except IntegrityError:
+                resource.metadata.update_element('rights', resource.metadata.rights.id,
+                                                 statement=self.rights.statement)
+        # if self.creation_date:
+        #     res_created_date = resource.metadata.dates.all().filter(type='created')[0]
+        #     resource.metadata.update_element('date', res_created_date.id, created=self.creation_date)
+        if self.language:
+            resource.metadata.update_element('language', resource.metadata.language.id,
+                                             code=self.language)
+        if self.modification_date:
+            res_modified_date = resource.metadata.dates.all().filter(type='modified')[0]
+            # This doesn't seem to be working.  Maybe updating an element triggers the mod. date to be
+            #  updated.
+            resource.metadata.update_element('date', res_modified_date.id, modified=self.modification_date)
```

Comment:
The creation date and modification date are always set to the current date which uses the following function calls: 

For update:

    resource.metadata.update_element('date', res_modified_date.id)

For create:

    resource.metadata.update_element('date', res_created_date.id)

If we need to be able to pass a date value in the above calls to use that instead of the current time, we have to change the update method of the Date metadata element class.